packages/astro/src/content/types-generator.ts

```diff
@@ -455,7 +455,10 @@ async function writeContentFiles({
 						settings.config.experimental.contentCollectionJsonSchema &&
 						collectionConfig?.schema
 					) {
-						let zodSchemaForJson = collectionConfig.schema;
+						let zodSchemaForJson =
+							collectionConfig.schema instanceof Function
```

Comment:
I think we usually check for functions this way:
```suggestion
							typeof collectionConfig.schema === 'function'
```